frontend/packages/console-dynamic-plugin-sdk/src/extensions/dashboards.ts

```diff
@@ -0,0 +1,469 @@
+    model: EncodedCodeRef;
+    /** Function which will map various statuses to groups. */
+    mapper?: EncodedCodeRef;
+    /** Additional resources which will be fetched and passed to `mapper` function. */
+    additionalResources?: EncodedCodeRef;
+    /** Defines whether the model's label or abbr should be used when rendering the item. Defaults to false (label). */
+    useAbbr?: boolean;
+    /** Loader for the component which will be used when item is expanded. */
+    expandedComponent?: EncodedCodeRef;
+  };
+
+  export type DashboardsOverviewInventoryItemCodeRefs = {
+    model: CodeRef<K8sKind>;
+    mapper?: CodeRef<StatusGroupMapper>;
+    additionalResources?: CodeRef<WatchK8sResources<any>>;
+    expandedComponent?: CodeRef<LazyLoader<ExpandedComponentProps>>;
+  };
+
+  /** Adds an item to the overview utilization card */
+  export type DashboardsOverviewUtilizationItem = {
+    /** The utilization item to be replaced */
+    id: string;
+    /** The Prometheus utilization query */
+    query: string;
+    /** The Prometheus total query */
+    totalQuery: string;
+  };
+
+  /** Adds an activity to the Activity Card of Overview Dashboard where the triggering of activity  is based on watching a K8s resource. */
+  export type DashboardsOverviewResourceActivity = {
+    /** The utilization item to be replaced */
+    k8sResource: EncodedCodeRef;
+    /**
+     * Function which will determine if given resource represents the action.
+     * If the function is not defined, every resource represents activity.
+     */
+    isActivity?: EncodedCodeRef;
+    /** Timestamp for given action, which will be used for ordering */
+    getTimestamp?: EncodedCodeRef;
+    /** Loader for corresponding action component */
+    loader: EncodedCodeRef;
+  };
+
+  export type DashboardsOverviewResourceActivityCodeRefs = {
+    k8sResource: CodeRef<FirehoseResource & { isList: true }>;
+    isActivity?: CodeRef<(resource: K8sResourceKind) => boolean>;
+    getTimestamp?: CodeRef<(resource: K8sResourceKind) => Date>;
+    loader: CodeRef<LazyLoader<K8sActivityProps>>;
+  };
+
+  /** Adds an activity to the Activity Card of Overview dashboard where the triggering of activity is based on the prometheus query watch. */
+  export type DashboardsOverviewPrometheusActivity = {
+    /** Queries to watch */
+    queries: string[];
+    /** Function which will determine if given query results represent the action */
+    isActivity: EncodedCodeRef;
+    /** Loader for corresponding action component */
+    loader: EncodedCodeRef;
+  };
+
+  export type DashboardsOverviewPrometheusActivityCodeRefs = {
+    isActivity?: CodeRef<(results: PrometheusResponse[]) => boolean>;
+    loader: CodeRef<LazyLoader<K8sActivityProps>>;
+  };
+
+  /** Adds an activity to the Activity Card of Overview dashboard where the triggering of activity is based on the prometheus query watch. */
+  export type ProjectDashboardInventoryItem = {
```

Comment:
this is the same as `DashboardsOverviewInventoryItem`. We should reuse the type

Reply:
the only difference is `useAbbr` which we dont use anyway? removing `ProjectDashboardInventoryItem`